modules/storage/cosmos-db/module/cosmos.bicep

```diff
@@ -0,0 +1,286 @@
+param name _name
+param location _location
+param kind _kind = 'GlobalDocumentDB'
+param properties _properties
+
+resource cosmosDB 'Microsoft.DocumentDB/databaseAccounts@2022-05-15' = {
+  name: name
+  location: location
+  kind: kind
+  properties: properties
+}
+
+output id string = cosmosDB.id
+output name _name = cosmosDB.name
+
+type CosmosDB = {
+  name: _name
+  location: _location
+  tags: _tags?
+  kind: _kind?
+  identity: _identity?
+  properties: _properties
+}
+
+@description('The resource name.')
+@minLength(3)
+@maxLength(64)
+type _name = string
+
+@description('The location of the resource group to which the resource belongs.')
+type _location = 'eastus' | 'eastus2' | 'westus' | 'westus2' | 'westcentralus' | 'westeurope' | 'southeastasia' | 'northeurope' | 'uksouth' | 'ukwest' | 'australiaeast' | 'australiasoutheast' | 'brazilsouth' | 'southindia' | 'centralindia' | 'westindia' | 'canadacentral' | 'canadaeast' | 'japaneast' | 'japanwest' | 'koreacentral' | 'koreasouth' | 'francecentral' | 'southafricanorth' | 'uaenorth' | 'switzerlandnorth' | 'germanywestcentral' | 'norwayeast'
```

Comment:
We cannot make location a type. The available locations depend on the Cloud environments the module is going to be deployed to and the SKU of the resources for some resource types. Besides, if ARM spins up a new location, all existing versions of the module will be broken.

Reply:
ill roll this out. Need to think about how to make location easier for the user to select, without also restricting their options.